### DISCUSSION TITLE
Secondary actions in Cards

### DISCUSSION THREAD

**Discussion by PatriRoman** [Telefonica/mistica-design#796] [Proposals]
Para poder implementar la funcionalidad de Wishlist necesitamos añadir a las media cards una opción de "añadir a favoritos", ya que la ficha de producto al no ser nuestra no podemos tocarla.

Dado que las cards a día de hoy ya están repletas de zonas de acción y carga cognitiva, sugerimos añadir al la zona de los buttons un menu kebap. El cual desplegará una bottom sheet en donde podremos añadir varias acciones, incluso quitar algunas que tenemos ahora mismo a nivel card y llevárnosla ahí, como por ejemplo "Términos y condiciones", "dismiss", etc. Y puede ser escalable si queremos añadir a futuro otro tipo de acciones como "compartir".

El problema principal se presenta en las cards que no tienen cta, que son todas clicables. En donde habría que añadir un área inferior en donde se encuentra el menú kebap que levante la bottom sheet:

<img width="1093" alt="image" src="https://user-images.githubusercontent.com/108683238/192458114-7b372233-34f6-4f93-90a1-cf806805e177.png">

El Figma donde podéis encontrar los diseños es [este](https://www.figma.com/file/LHob8XrzUxozcET7i9shTz/UX-Flow-Wishlist-Test?node-id=1218%3A21870).

**Comment by AnaMontes11** [comment]
**Tenemos una petición similar por parte de Vivo:**

El equipo de Vivo tiene la necesidad de añadir la acción de agregar favoritos en la Media Card. Para ello, comentan si es posible cambiar la acción del botón cerrar a la acción de favorito (corazón).

![vivo](https://user-images.githubusercontent.com/88187691/192475528-b71369bf-5d74-4c3a-b2f4-668f9dce6a80.jpg)

**Reply by PatriRoman** [reply]
¿Entonces en este caso podría darse a favorito pero perderíamos la opción de dismiss?

Veo el mismo inconveniente que en las otras: en las cards que no hay cta...¿cómo lo solucionamos? ¿un área clicable dentro de la imagen?

**Reply by yceballost** [reply]
si fuesemos por esta posibilidad yo creo que si, que o usas el dismiss o usas la otra acción. Me convence más tu propuesta porque de esa manera no mezclamos acciones con ese dismiss

**Comment by yceballost** [comment]
Ticket creado para trabajar en esto
https://github.com/Telefonica/mistica-design/issues/832

**Comment by yceballost** [comment]
Ha llovido mucho por aquí.

Con la creación de las nuevas display cards hemos decidido unificar este area de "alternative actions" en las Display Media, Display Data, Media card y data card. Todas estas tipologías de cards tirarían del mismo componente de "top actions"

Estas top actions podrán llevar máximo 2 acciones siendo la siguiente lista todas las posibilidades que admitiría estas zona:
- Action 1 + Action 2
- Action 1 + Dismiss
- Action 1
- Dismiss

Action 1 puede ser cualquier icono con acción, por ejemplo: 
- un icono de un corazón para marcar como favorito, 
- un icono de kebab que despliega una bottom sheet o una action sheet para mostrar más acciones
- ...

![image](https://user-images.githubusercontent.com/6722153/213028257-8942a930-89b0-48f1-92fc-f61977537101.png)
